scripts/tests/chiptest/linux.py

```diff
@@ -197,6 +199,19 @@ def terminate(self):
         for command in self.COMMANDS_TERMINATE:
             self.run(command)
 
+    def wrap_in_namespace(self, subproc: SubprocessInfo):
+        return subproc.wrap_with("ip", "netns", "exec", "{}-{}".format(subproc.kind, self.index))
+
+
+class LinuxNamespacedExecutor(Executor):
+    def __init__(self, ns):
+        self.ns = ns
+
+    def run(self, subproc: SubprocessInfo, stdin, stdout, stderr):
```

Comment:
I would default `stdin`, `stdout`, etc. to the default value used by `subprocess.Popen`, but better yet, I would create API similar to `asyncio.create_subprocess_exec` where there is also `**kw` which is passed directly to `Popen()`.

Reply:
The `run` function needs to get a subprocess info object because the `kind` information needs to be passed to it implement namespaced runners. `stdin` et al. improved.